sa/sa_test.go

```diff
@@ -4565,3 +4567,346 @@ func TestUpdateRegistrationKey(t *testing.T) {
 		})
 	}
 }
+
+type mockRLOStream struct {
+	grpc.ServerStream
+	sent []*sapb.RateLimitOverride
+	ctx  context.Context
+}
+
+func newMockRLOStream() *mockRLOStream {
+	return &mockRLOStream{ctx: ctx}
+}
+func (m *mockRLOStream) Context() context.Context { return m.ctx }
+func (m *mockRLOStream) RecvMsg(any) error        { return io.EOF }
+func (m *mockRLOStream) Send(ov *sapb.RateLimitOverride) error {
+	m.sent = append(m.sent, ov)
+	return nil
+}
+
+func TestAddRateLimitOverrideInsertThenUpdate(t *testing.T) {
+	if os.Getenv("BOULDER_CONFIG_DIR") != "test/config-next" {
```

Comment:
Should we add a TODO here (and on the other new conditionals like it) to make sure to remove it once enabled in prod?